### PR TITLE
Add a guide on building an image for JH and adding it to ODH.

### PR DESCRIPTION
**Pull request by martinpovolny**
Converting this issue
https://github.com/operate-first/odh-moc-support/issues/2
 into a guide

review here: https://github.com/martinpovolny/data-science-workflows/blob/image_create_deploy/docs/create_and_deploy_jh_image.md

### CODE REVIEW

**Comment by martinpovolny** [comment]
@MichaelClifford , @tumido , @oindrillac : Please, review.

File: docs/create_and_deploy_jh_image.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
+# How to create a Jupyterhub image and deploy it on ODH
+
+This howto is about getting your repository built into an image and having that image made available in (your) Jupyterhub instance.
+
+We are going to deal with 2 issues:
+
+ 1. Have an image build and added to a public image repository.
```
Comment (oindrillac) [inline docs/create_and_deploy_jh_image.md:7]: Can you modify this to - "image built" if build is used as a verb or 
" Have an image build and get it added ..." if image build is a noun here

File: docs/create_and_deploy_jh_image.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+ 1. Add your image to an ODH instance (that has a Jupyterhub in it).
+
+
+If you don't have an ODH instance that you would like to work with, you can create one using these quides:
```
Comment (oindrillac) [inline docs/create_and_deploy_jh_image.md:11]: Change to "guides"

File: docs/create_and_deploy_jh_image.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+Here's an example that goes together with the above example of an ImageStream:
+
+https://github.com/aicoe-aiops/categorical-encoding/blob/master/.aicoe-ci.yaml
+
+I am going to compile this into a guide and would be grateful for your testing of the process or improvement suggestions.
+
+Information on enabling Thoth on your repo is here: https://github.com/AICoE/aicoe-ci#setting-aicoe-ci-on-github-organizationrepository
+
+For 3rd party project outside of AICoe one shall change the value of `registry-org` (quay.io org) to to his own value.
```
Comment (oindrillac) [inline docs/create_and_deploy_jh_image.md:27]: Modify to "AICoE"

File: docs/create_and_deploy_jh_image.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+
+![Setting up CI integration with Thoth](./../public/assets/jh_image_howto/ci-integration.png)
+
+If your project is under the `aicoe-aiops` (and some other) GH orgs then this is already done on the org level and you don't need it.
+
+### Setting up the image repository
+
+Next you need to have a place to upload the built images. For that you need an account and a project on quay.io.
+
+Again there already exists an 'AiCOE' organization in Quay: https://quay.io/organization/aicoe so if your project is part of AICOE you can use that.
```
Comment (oindrillac) [inline docs/create_and_deploy_jh_image.md:47]: small nit - but let's consistently spell AICoE.

File: docs/create_and_deploy_jh_image.md [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+Thoth needs to eb able to access your project -- push images into it. That can be accomplished by creating a robot account with a push (pull) secret.
```
Comment (oindrillac) [inline docs/create_and_deploy_jh_image.md:49]: change to "be"

File: docs/create_and_deploy_jh_image.md [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+
+There are pre-configured push secrets for various projects already present in Thoth. See the bottom of this page: https://github.com/AICoE/aicoe-ci#services-and-features.
+
+If your project is in one of these, you can skip this step.
+
+
+![Quay pull secret setup](../public/assets/jh_image_howto/quay-robot.png)
+
+After creating the robot you need to set the permissing it will possess.
```
Comment (oindrillac) [inline docs/create_and_deploy_jh_image.md:58]: do you mean "permissions"

File: docs/create_and_deploy_jh_image.md [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+
+If you have done these steps correctly, Thoth will pick up your tag and run a image build for you.
+
+## Checking and debugging the build
+
+You can login to https://tekton-dashboard-openshift-pipelines.apps.ocp4.prod.psi.redhat.com/ and find your build by using the search field. Enter "project:<YOUR_REPO_NAME>".
+
+![Tekton: search a build](../public/assets/jh_image_howto/thoth-pipeline-run.png)
+
+The part that in relevant for image building is under "tag-release-run". There may by other sections that are red, but if this part is green the image was build and pushed. You can also check that on Quay.io.
```
Comment (oindrillac) [inline docs/create_and_deploy_jh_image.md:112]: change to "that is"

File: docs/create_and_deploy_jh_image.md [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+
+# Adding the image where you need it
+
+Once you image is built (check in Quay), you can use it in your instance of ODH.
```
Comment (oindrillac) [inline docs/create_and_deploy_jh_image.md:116]: change to "your"

File: docs/create_and_deploy_jh_image.md [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+        importPolicy:
+          scheduled: true
+  ```
+
+The important part is:
+```
+  labels:
+    opendatahub.io/notebook-image: "true"
+```
+This tells JuputerHub that it has to offer this particular image then creating an instance for it's users.
```
Comment (oindrillac) [inline docs/create_and_deploy_jh_image.md:147]: Change to "JupyterHub"

**Review by oindrillac — CHANGES_REQUESTED** [review]
Thanks for putting this together @martinpovolny 🙂 
I added a few comments for some minor changes

**Comment by martinpovolny** [comment]
Thanks @oindrillac. I have fixed all the spelling as you suggested.

File: docs/create_and_deploy_jh_image.md [hunk 10]
```diff
@@ -0,0 +1,160 @@
+# How to create a Jupyterhub image and deploy it on ODH
+
+This howto is about getting your repository built into an image and having that image made available in (your) Jupyterhub instance.
+
+We are going to deal with 2 issues:
+
+ 1. Have an image built and added to a public image repository.
+ 1. Add your image to an ODH instance (that has a Jupyterhub in it).
```
Comment (oindrillac) [inline docs/create_and_deploy_jh_image.md:8]: I missed that it's sr no. 1. here. Can you change that to 2 please?

Comment (martinpovolny) [reply docs/create_and_deploy_jh_image.md:8]: writing it as "1." every time triggers the automatic numbering feature of markdown

so this in intended and it correctly renders as "1." and "2." (see the link in description)

File: docs/create_and_deploy_jh_image.md [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+### Adding configuration file to your repo
+
+Next we need to finish configuration of the source repo. This is done by adding a config file named `.aicoe-ci.yaml`.
+
+
+Here's an example of a working minimal config: https://github.com/aicoe-aiops/project-template/blob/master/.aicoe-ci.yaml
+
+Uncomment the `build` section to enable Thoth build.
+
+The builder can do a number of things. To have the S2I correctly built these 2 lines are essential:
```
Comment (MichaelClifford) [inline docs/create_and_deploy_jh_image.md:79]:
```suggestion
The builder can do a number of things. To have the S2I correctly build these 2 lines are essential:
```

File: docs/create_and_deploy_jh_image.md [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+
+If you have done these steps correctly, Thoth will pick up your tag and run a image build for you.
+
+## Checking and debugging the build
+
+You can login to https://tekton-dashboard-openshift-pipelines.apps.ocp4.prod.psi.redhat.com/ and find your build by using the search field. Enter "project:<YOUR_REPO_NAME>".
+
+![Tekton: search a build](../public/assets/jh_image_howto/thoth-pipeline-run.png)
+
+The part that is relevant for image building is under "tag-release-run". There may by other sections that are red, but if this part is green the image was build and pushed. You can also check that on Quay.io.
```
Comment (MichaelClifford) [inline docs/create_and_deploy_jh_image.md:112]:
```suggestion
The part that is relevant for image building is under "tag-release-run". There may by other sections that are red, but if this part is green the image was built and pushed. You can also check that on Quay.io.
```

**Review by MichaelClifford — CHANGES_REQUESTED** [review]

**Comment by martinpovolny** [comment]
Updated with suggested fixes.

File: docs/create_and_deploy_jh_image.md (hunk 10 above)
Comment (oindrillac) [reply docs/create_and_deploy_jh_image.md]: Got it! @martinpovolny 🙂

**Review by oindrillac — APPROVED** [review]
LGTM 🙂👍

**Review by MichaelClifford — APPROVED** [review]
LGTM

**Pull request merged**